src/games/strategy/triplea/ui/MapRouteDrawer.java

```diff
@@ -106,9 +101,11 @@ private void drawJoints(final Graphics2D graphics, final Point[] points, final i
     final int jointsize = 10;
     // If the points array is bigger than 1 the last joint should not be drawn (draw an arrow instead)
     final Point[] newPoints = points.length > 1 ? Arrays.copyOf(points, points.length - 1) : points;
-    for (final Point p : newPoints) {
-      graphics.fillOval((int) (((p.x - xOffset) - (jointsize / 2) / scale) * scale),
-          (int) (((p.y - yOffset) - (jointsize / 2) / scale) * scale), jointsize, jointsize);
+    for (Point[] joints : routeOptimizer.getAllPoints(newPoints)) {
```

Comment:
Some commentary on what is happening will help us maintain this in the future.
Ideally this would be broken up, possibly extracted to a new method or class etc.. Minimum bar IMO is something to help a developer reading this to know what is going on.